pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go

```diff
@@ -148,6 +149,11 @@ func installer(allWorkers bool, secretName string) []runtime.Object {
 		})
 	}
 
+	env = append(env, corev1.EnvVar{
```

Comment:
Ideally we don't want this ENV at all if `redeploySystemAgentGeneration` is not set because it will result in a change of checksum and redeployment of system-agent if this change gets applied to cluster that was generated before the change. I know we never released this functionality, but it would be cleaner anyway to not add this env if it's not set on the Cluster CR. With current code I believe it would be set to 0.

Reply:
yeah, I can change that.